tests/kernel/timer/timer_api/src/main.c

```diff
@@ -18,6 +19,7 @@ struct timer_data {
 #define EXPIRE_TIMES 4
 #define WITHIN_ERROR(var, target, epsilon)       \
 		(((var) >= (target)) && ((var) <= (target) + (epsilon)))
+#define INEXACT_MS_CONVERT (k_ticks_to_ms_ceil64(k_ms_to_ticks_ceil64(1)) != 1)
```

Comment:
This might be a little more robust if you checked that they weren't divisible directly.  Something like:

```c
#define INEXACT_MS_CONVERT                       \
    (MAX(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,     \
         CONFIG_SYS_CLOCK_TICKS_PER_SEC) %       \
     MIN(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,     \
         CONFIG_SYS_CLOCK_TICKS_PER_SEC) == 0)   \
```

Honestly I had to think really hard to figure out if that double conversion was guaranteed to detect the case you want, and I'm still not completely convinced it does for all non-divisible rates.